lib/Doctrine/ORM/EntityManager.php

```diff
@@ -156,6 +164,7 @@ protected function __construct(Connection $conn, Configuration $config, EventMan
         $this->metadataFactory->setEntityManager($this);
         $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
 
+        $this->persisterFactory  = new PersisterFactory($this);
```

Comment:
There's a way to fix this, if we introduce 2 concepts:
* EntityGraph which would drastically simplify UoW by being part of another class ChangeSet that tracks Entity states.
* EntityBinding which would store a list of Metadata, Persister and have access to a given Entity's identifiers.

If I could traverse between EntityBindings to get Metadata of others, its Persisters inside of an existing Persister, then yes, we could circle back these dependency hell. But it would introduce some serious cost in performance. As I said in the beginning, I'm proposing to do the bare minimum to fully support many to many extra lazy operations.
The PersisterFactory is required because a subsequent PR will be responsible to generate SQL pieces, allowing to address the currently impossible bug #1033 . This SQL class would be instantiated inside of PersisterFactory and shared between Persisters. That's the only way I found to do it without entering on a Refactoring hell of UoW, Persisters and small bits of Hydrators.